### PR TITLE
Ors client

### PR DESCRIPTION
**Pull request by osundwajeff**
This class `ORSClient` uses ORS tools API directly instead of using the plugin in QGIS.

It is accompanied by a unittest.

#261 will use this client to be able to compute multibuffers used service area analysis.

### CODE REVIEW

File: geest/core/ors_client.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import os
+from PyQt5.QtCore import QUrl, QByteArray
```
Comment (timlinux) [inline geest/core/ors_client.py:2]: Please import pyqt via the pyqgis api. See my code for examples

File: geest/core/ors_client.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        url = f"{self.base_url}/{endpoint}"
+        request = QNetworkRequest(QUrl(url))
+
+        # Set necessary headers for the ORS API
+        request.setHeader(QNetworkRequest.ContentTypeHeader, "application/json")
+        if self.api_key:
+            request.setRawHeader(b"Authorization", self.api_key.encode())
+        else:
+            QgsMessageLog.logMessage(
+                "API Key is missing", "ORS", QgsMessageLog.CRITICAL
```
Comment (timlinux) [inline geest/core/ors_client.py:30]: Qgis.Critical?

File: geest/core/ors_client.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        self.api_key = os.getenv("ORS_API_KEY")
+
+        # Ensure the API key is available
+        if not self.api_key:
+            raise EnvironmentError(
+                "ORS API key is missing. Set it in the environment variable 'ORS_API_KEY'."
+            )
+
+    def make_request(self, endpoint, params):
+        url = f"{self.base_url}/{endpoint}"
```
Comment (timlinux) [inline geest/core/ors_client.py:21]: QgsNetworkRequest?

File: geest/core/ors_client.py (hunk 2 above)
Comment (osundwajeff) [reply geest/core/ors_client.py:30]: Yes, it should be that

File: geest/core/ors_client.py (hunk 3 above)
Comment (osundwajeff) [reply geest/core/ors_client.py:21]: QgsNetworkAccessManager uses `QNetworkRequest` as input

**Pull request merged**